Library/OcDeviceMiscLib/SetResizableBar.c

```diff
@@ -309,8 +377,241 @@ SetResizableBarOnDevice (
+    // Bit 1 is set: supports operating with the BAR sized to 2 MB
+    // Bit n is set: supports operating with the BAR sized to (2^n) MB
+    //
+    // Reference values for RX 6900 with two resizable BARs.
+    // Disabled values:
+    //   Resizeable Bar Capability [1]
+    //     ResizableBarCapability         0007F000
+    //     ResizableBarControl            0840
+    //   Resizeable Bar Capability [2]
+    //     ResizableBarCapability         00001FE0
+    //     ResizableBarControl            0102
+    // Enabled values:
+    //   Resizeable Bar Capability [1]
+    //     ResizableBarCapability         0007F000
+    //     ResizableBarControl            0E40
+    //   Resizeable Bar Capability [2]
+    //     ResizableBarCapability         00001FE0
+    //     ResizableBarControl            0802
+    //
+    NewCapabilities = Capabilities = LShiftU64 (Entries[Index].ResizableBarControl.Bits.BarSizeCapability, 28)
+                                     | Entries[Index].ResizableBarCapability.Bits.BarSizeCapability;
+
+    //
+    // Restrict supported BARs to specified value.
+    //
+    NewCapabilities &= PCI_BAR_CAP_LIMIT (Size);
+
+    //
+    // Disable bits higher than current as we are not allowed to increase bar size
+    // more than we already have.
+    //
+    if (!Increase) {
+      NewCapabilities &= PCI_BAR_CAP_LIMIT (Entries[Index].ResizableBarControl.Bits.BarSize);
+    }
+
+    //
+    // If requested BAR size is too low, choose the lowest available BAR size.
+    //
+    if (  (NewCapabilities == 0)
+       && (Entries[Index].ResizableBarControl.Bits.BarSize > (UINT32)Size))
+    {
+      Bit = LowBitSet64 (Capabilities);
+    } else {
+      Bit = HighBitSet64 (NewCapabilities);
+    }
+
+    DEBUG ((
+      DEBUG_INFO,
+      "OCDM: RBAR %u/%u supports 0x%Lx, sizing %u inc %d results setting from %u to %d\n",
+      Index + 1,
+      ResizableBarNumber,
+      Capabilities,
+      Size,
+      Increase,
+      Entries[Index].ResizableBarControl.Bits.BarSize,
+      (INT32)Bit
+      ));
+
+    //
+    // If we have no supported configuration, just skip.
+    //
+    if ((Bit < 0) || (Entries[Index].ResizableBarControl.Bits.BarSize == (UINT32)Bit)) {
+      continue;
+    }
+
+    Offset = ResizableBarOffset + sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_HEADER)
+             + Index * sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_ENTRY)
+             + OFFSET_OF (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_ENTRY, ResizableBarControl);
+
+    Entries[Index].ResizableBarControl.Bits.BarSize = (UINT32)Bit;
+    PciRootBridgeIo->Pci.Write (
+                           PciRootBridgeIo,
+                           EfiPciWidthUint32,
+                           PciAddrOffset (PciAddress, Offset),
+                           1,
+                           &Entries[Index].ResizableBarControl.Uint32
+                           );
+
+    ChangedBars = TRUE;
+  }
+
+  if (ChangedBars) {
```

Comment:
This is only ever set to TRUE above, so why have a var and a check?

Reply:
This also is there in the PciIo version fixing it anyways though